### ISSUE TITLE
Dependency cycle error if sources are managed exclusively by puppet

### ISSUE THREAD

**Issue by mickaelperrin** [voxpupuli/puppet-unattended_upgrades#28]
### Description

If you use the `purge` option of the main `::apt` class, and want to manage sources exclusively in Puppet, then you fall in a dependency cycle error when using this module.

### Problem

The sources are purged, they need to be updated, but they wait for apt::conf of this module and apt::conf require the Package['unattended-upgrades'] that can't be installed because sources are purged or wait for  Class['apt::update'] .

### Testing code
```puppet
    class base::apt inherits base {
    
      $ubuntu_location = 'http://archive.ubuntu.com/ubuntu'
      $ubuntu_repos = 'main universe'
    
      class { 'apt':
        purge => {
          'sources.list'   => true,
          'sources.list.d' => true,
          'preferences'    => true,
          'preferences.d'  => true
        },
      }
    
      apt::source { 'ubuntu':
        location    => $ubuntu_location,
        release     => $lsbdistcodename,
        repos       => $ubuntu_repos,
        include_src => true,
      }
    
      apt::source { 'ubuntu_updates':
        location    => $ubuntu_location,
        release     => "${lsbdistcodename}-updates",
        repos       => $ubuntu_repos,
        include_src => true,
      }
    
      apt::source { 'ubuntu_security':
        location    => $ubuntu_location,
        release     => "${lsbdistcodename}-security",
        repos       => $ubuntu_repos,
        include_src => true,
      }
    }
```
My `base` class includes many modules and some of them use also the `::apt` class, like the docker one [garethr-docker](https://github.com/garethr/garethr-docker/blob/master/manifests/repos.pp) and the unattended-upgrades.

So, I need to execute apt_update before installating anything.

    class base {
       Class['apt::update'] -> Package<| |>
        include 'base::apt'
        include 'unattended-upgrades'
        include 'base::install' # list of packages to install
    }

Then my `node` is simply:

    node default {
        include 'base'
    }

**Comment by mickaelperrin** [comment]
Removing requirements on Package['unattended-upgrades'] resolved the issue. 

See this [commit](https://github.com/mickaelperrin/puppet-unattended_upgrades/commit/3141fcf8ba4ad4b1ef2cfab68ae78e66d747e52e).

But maybe there is a better way to go ?

**Comment by ffrank** [comment]
How about

    Class['apt::update'] -> Package<| tag != 'unattended-upgrades' |>

That doesn't scale (I think), but then the catch-all collector is a dangerous beast in any case.

**Comment by daenney** [comment]
> The sources are purged, they need to be updated, but they wait for apt::conf of this module and apt::conf require the Package['unattended-upgrades'] that can't be installed because sources are purged or wait for Class['apt::update'] .

As far as I can see this would only be a problem if you've explicitly defined `Class['::apt::update'] -> Package <||>`. It wouldn't cause a dependency cycle just because of the package requirement.

**Comment by daenney** [comment]
I think this is fixed as a side effect of 67ee9eca5749fc54d8060cb7b87c8bd56c20765c. Since we no longer notify `Apt::Update` by default (as that was rather pointless) it should have resolved this dependency cycle.

**Comment by jyaworski** [comment]
Closing due to the above commit.
